File: testprojects/csharp/benchmark/presentation/gui/stumbleupon/QuestionGUI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CSharpBenchmark.domain.stumbleupon;

namespace CSharpBenchmark.presentation.gui.stumbleupon
{

    //Functional requirement 3.2.2
    //Test case 165: All classes in package presentation.gui.stumbleupon are not allowed to use in a not direct lower layer
    //Result: TRUE

    //FR5.6
    [StumbleAnnotation(stumbleId = "1234567890")]
    public class QuestionGUI
    {
    }
}

File: testresources/java/benchmark_2012/presentation/gui/facebook/QuestionGUI.java
package presentation.gui.facebook;

import domain.facebook.FacebookAnnotation;

//Functional requirement 3.1.2
//Test case 84: Only class presentation.gui.facebook.QuestionGUI may have a dependency with domain.facebook.FacebookAnnotation
//Result: FALSE

//FR5.6
@FacebookAnnotation(facebookId = "1234567890")
public class QuestionGUI {
}